tools/cli/internal/versioning/version.go

```diff
@@ -0,0 +1,94 @@
+// Copyright 2024 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package versioning
+
+import (
+	"fmt"
+	"regexp"
+	"time"
+)
+
+type APIVersion struct {
```

Comment:
if this is your "main" struct it make sthe case for the package to be called `version` or `apiversion`

Reply:
and then what should be the name of the struct? is it ok to make it apiversion.APIVersion? or something else like T? (openapi3.T as an example)

Reply:
it's ok to stutter on the struct name apiversion.APIVersion is okish, again this is all ok to play with things like
`apiversion.Version`, `version.Version`, `version.APIVersion` tho traditionally your "main" struct, will match the package name, and same as with `version.New` make sure that any combination like `version.Parse` make sense when put together, this is usually the best way to know your name, when consuming your public methods does it look ok when combining package and methods